addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2217,18 +2220,18 @@ export class OdooEditor extends EventTarget {
             // Remove all illegal attributes and classes from the node, then
             // clean its children.
             for (const attribute of [...node.attributes]) {
-                // we kee some style on span to be able to paste text styled in the editor
-                if (node.nodeName === 'SPAN' && attribute.name === 'style') {
-                    const spanInlineStyles = attribute.value.split(';');
+                // Keep allowed styles on nodes with allowed tags.
+                if (CLIPBOARD_WHITELISTS.styledTags.includes(node.nodeName) && attribute.name === 'style') {
+                    const spanInlineStyles = attribute.value.split(';').map(x => x.trim());
                     const allowedSpanInlineStyles = spanInlineStyles.filter(rawStyle => {
                         const [styleName, styleValue] = rawStyle.split(':');
-                        const style = CLIPBOARD_WHITELISTS.spanStyle[styleName.trim()];
+                        const style = CLIPBOARD_WHITELISTS.styles[styleName.trim()];
                         return style && !style.defaultValues.includes(styleValue.trim());
                     });
                     node.removeAttribute(attribute.name);
                     if (allowedSpanInlineStyles.length > 0) {
                         node.setAttribute(attribute.name, allowedSpanInlineStyles.join(';'));
-                    } else {
+                    } else if (['SPAN', 'FONT'].includes(node.tagName)) {
```

Comment:
Why the `else` here ?
Do we assume that 'SPAN' and 'FONT' tags never come with allowed inline style attributes?

Reply:
For the record, I replied to this comment on Discord.
> If I recall correctly the else is there so that if the span or font does not have any allowed style then we remove it.
>If it has allowed styles then we purge the disallowed by re-setting the style attribute on the line just above
> but if it has no allowed styles then it is no use to set the style attribute to an empty string so we just remove the span or font